api2-war/src/main/java/eu/europeana/api2/v2/service/translate/MetadataTranslationService.java

```diff
@@ -54,36 +57,46 @@ public MetadataTranslationService(TranslationApiClient translationApiClient, Met
         long start = System.currentTimeMillis();
-
-        String chosenLanguage = metadataChosenLanguageService.getMostRepresentativeLanguageForSearch(beans, targetLanguage);
-        LOG.debug("Most representative language chosen for translations is  {}", chosenLanguage);
-
-        // if there is no chosen language stop the translation workflow OR
-        // If the chosen language matches target language then do nothing (stop the workflow)
-        // keep in mind during Ingestion we always translate the record to english. So target language is always en
-        if (chosenLanguage == null || StringUtils.equals(chosenLanguage, targetLanguage)) {
-            LOG.debug("Stop the translation workflow for search results ..." );
-            return beans;
-        }
-        TranslationMap textToTranslate = new TranslationMap(chosenLanguage);
+        List<TranslationMap> textsToTranslate = new ArrayList<>();
 
         int index = 0;
         for (BriefBean bean : beans) {
             LOG.debug("Check search result {}...", index);
             int finalIndex = index;
-            ReflectionUtils.doWithFields(bean.getClass(), field ->  getSearchValueToTranslateForField(bean, field, chosenLanguage, textToTranslate, finalIndex), BaseService.searchFieldFilter);
+            String chosenLanguage = metadataChosenLanguageService.getMostRepresentativeLanguage(bean, targetLanguage, true);
+            LOG.debug("Most representative language chosen for {} search result is  {}", index, chosenLanguage);
+
+            // if there is no chosen language stop the translation workflow OR
+            // If the chosen language matches target language then do nothing
+            if (chosenLanguage == null || StringUtils.equals(chosenLanguage, targetLanguage)) {
+            LOG.debug("Stop the translation workflow for search result {} ...", index );
+            } else {
+                // gather texts for translations
+                TranslationMap gatheredTextForTranslation= new TranslationMap(chosenLanguage);
+                ReflectionUtils.doWithFields(bean.getClass(), field ->  getSearchValueToTranslateForField(bean, field, chosenLanguage, targetLanguage, gatheredTextForTranslation, finalIndex), BaseService.searchFieldFilter);
+                addGatheredTranslations(gatheredTextForTranslation, textsToTranslate, chosenLanguage);
+            }
             index++;
         }
 
         // if no translation gathered return
-        if (textToTranslate.isEmpty()) {
+        if (textsToTranslate.isEmpty()) {
             LOG.debug("No values gathered for translations. Stopping the translation workflow for search results");
             return beans;
         }
-        LOG.debug("Text to translate - {}", textToTranslate);
+
+        textsToTranslate.stream().forEach(value -> LOG.debug("Text to translate - {} : {}", value.getSourceLanguage(), value));
```

Comment:
Better wrap this in `if LOG.isDebugEnabled` for better performance in production.